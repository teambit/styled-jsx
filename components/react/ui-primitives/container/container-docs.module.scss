.blueBorder {
    border-color: #5353a7;
}