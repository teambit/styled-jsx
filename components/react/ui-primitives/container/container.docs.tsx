export default function Overview() {
    return null;
};

Overview.abstract = 'An imperfect container.';

Overview.labels = ['react', 'typescript', 'container', 'ui'];