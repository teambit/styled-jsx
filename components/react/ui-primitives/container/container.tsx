import React, {ReactElement} from 'react';
import cs from 'classnames'

export type ContainerProps = {
    /** The container content. */
    children: any,
    /** A className for customization */
    className?: string;
    /** Determines whether to style the container as "floating" */
    floatEffect?: boolean
}

export const Container = ({className, children, floatEffect = true} : ContainerProps) => {
    return(
    <div>
        <style jsx>{`
          .container {
            border: solid 7px #41403e;
            border-radius: 255px 15px 225px 15px/15px 225px 15px 255px;
            line-height: 1.5;
            padding: 25px;
            font-family: 'Indie Flower', cursive;
            font-size: 22px;
            max-width: 500px;
            }

            .float {
                box-shadow: 20px 38px 34px -26px hsla(0,0%,0%,.2);
                transition: box-shadow 0.5s;
            }

            .float:hover  {
                    box-shadow: 2px 8px 4px -6px hsla(0, 0%, 0%, 0.3);
            }
        `}</style>
        <div className={cs('container', floatEffect ? 'float' : null, className)}>
            {children}
        </div>   
     </div>
    )
}