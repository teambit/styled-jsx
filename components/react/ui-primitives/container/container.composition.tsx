import React from 'react'
import { Container } from './container'
import  styles  from './container-docs.module.scss'

export const FloatingContainer = () => {
    return (
        <Container>
            A "floating" container.
        </Container>
    )
}

export const GroundedContainer = () => {
    return (
        <Container floatEffect={false}>
            A "grounded" container.
        </Container>
    )
}


export const CustomizedContainer = () => {
    return (
        <Container className={styles.blueBorder} floatEffect={false}>
            A customized container.
        </Container>
    )
}