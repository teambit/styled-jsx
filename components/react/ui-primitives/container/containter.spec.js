import React from 'react';
import renderer from 'react-test-renderer';
import { Container } from './container';

it('renders correctly', () => {
    const element = renderer.create(<Container>Test Container</Container>).toJSON();
    expect(element).toMatchSnapshot();
  });