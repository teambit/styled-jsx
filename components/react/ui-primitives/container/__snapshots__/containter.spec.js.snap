// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <style
    global={true}
    jsx={true}
  >
    
            @import url('https://fonts.googleapis.com/css2?family=Indie+Flower&display=swap');
        
  </style>
  <style
    jsx={true}
  >
    
          .container {
            border: solid 7px #41403e;
            border-radius: 255px 15px 225px 15px/15px 225px 15px 255px;
            line-height: 1.5;
            padding: 25px;
            font-family: 'Indie Flower', cursive;
            font-size: 22px;
            max-width: 500px;
            }

            .float {
                box-shadow: 20px 38px 34px -26px hsla(0,0%,0%,.2);
                transition: box-shadow 0.5s;
            }

            .float:hover  {
                    box-shadow: 2px 8px 4px -6px hsla(0, 0%, 0%, 0.3);
            }
        
  </style>
  <div
    className="container float"
  >
    Test Container
  </div>
</div>
`;
